src/pke/lib/encoding/ckkspackedencoding.cpp

```diff
@@ -494,71 +495,76 @@ bool CKKSPackedEncoding::Decode(size_t noiseScaleDeg, double scalingFactor, enum
 
     double logstd = std::log2(stddev);
 
-    // if stddev < sqrt{N}/8 (minimum approximation error that can be achieved)
-    if (stddev < 0.125 * std::sqrt(GetElementRingDimension())) {
-        stddev = 0.125 * std::sqrt(GetElementRingDimension());
+    if (executionMode == EXEC_NOISE_ESTIMATION) {
+        m_logError = logstd;
     }
+    else {
+        // if stddev < sqrt{N}/8 (minimum approximation error that can be achieved)
```

Comment:
move this code in front of the if branch

Reply:
Do you mean the code setting stddev = 0.125 * sqrt(ringDim)? Should this be before we set logstd? It wasn't before it previously.

Reply:
Talked and decided not to do.